packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -126,6 +126,16 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
     const nuxtApp = useNuxtApp()
     const writeFinalCookieValue = () => {
       if (opts.readonly || isEqual(cookie.value, cookies[name])) { return }
+      nuxtApp._cookies ||= {}
+      if (name in nuxtApp._cookies) {
+        // do not append a second `set-cookie` header
+        if (isEqual(cookie.value, nuxtApp._cookies[name])) { return }
```

Comment:
Checking equality only allows dedup when exact same options are used. Cookies are distinct by name, domain, path, secure, httpOnly and sameSite.

Also parse from cookie-es, does not parse all `set-cookie` options

See https://github.com/unjs/h3/blob/b078181bd5c103dd7365bd33e263c492399afd67/src/utils/cookie.ts#L185, https://github.com/unjs/h3/blob/b078181bd5c103dd7365bd33e263c492399afd67/src/utils/cookie.ts#L44 and https://github.com/unjs/cookie-es/pull/43 for test.

We might do same thing in Nuxt composable or at least until adopting h3 v2 which properly do this.